streams/src/main/java/org/apache/kafka/streams/state/internals/MergedSortedCacheSessionStoreIterator.java

```diff
@@ -45,8 +47,8 @@
 
     @Override
     Windowed<Bytes> deserializeCacheKey(final Bytes cacheKey) {
-        final byte[] binaryKey = cacheFunction.key(cacheKey).get();
-        final byte[] keyBytes = SessionKeySchema.extractKeyBytes(binaryKey);
+        final ByteBuffer binaryKey = cacheFunction.key(cacheKey);
+        final ByteBuffer keyBytes = SessionKeySchema.extractKeyBytes(binaryKey);
         final Window window = SessionKeySchema.extractWindow(binaryKey);
```

Comment:
This is a good example that we avoided unnecessary array copy.
Before:
1. `cacheFunction.key(cacheKey)`  -> do an array copy
2. `SessionKeySchema.extractKeyBytes(binaryKey)` -> do an array copy

After:
1. `cacheFunction.key(cacheKey)`  -> no array copy, just return sliced byte buffer
2. `SessionKeySchema.extractKeyBytes(binaryKey)` -> no array copy, just return sliced byte buffer